docs/shell_setup.md

```diff
@@ -0,0 +1,180 @@
+
+!!! note "ZSH deprecation warning"
+    To get rid of the deprecation warning in OSX, export this variable in your shell. Add it to your `.zshrc` file for persistance:
+
+    ```bash linenums="1" title="~/.zshrc"
+    export BASH_SILENCE_DEPRECATION_WARNING=1
+    ```
+### Oh My ZSH !
+
+Oh My Zsh is a delightful, open source, community-driven framework for managing your Zsh configuration. It comes bundled with thousands of helpful functions, helpers, plugins, themes, and a few things that make you shout... `"Oh My ZSH!"`
+
+#### install
+
+```bash
+sh -c "$(curl -fsSL https://raw.github.com/ohmyzsh/ohmyzsh/master/tools/install.sh)"
+```
+
+#### Theme
+
+There are many great themes out there to help developers to become more productive and make the Terminal look cool.
+On such theme is `PowerLevel10k`. It's easy to install and provides a ton of tooling in your shell. It's available in a git repo that you have to clone in your `.oh-my-zsh`  custom folder.
+
+!!! note "Fonts"
+    When using iTerm2 or Termux, p10k configure can install the recommended font automatically. Simply answer Yes when asked whether to install Meslo Nerd Font.
+
+    Else, [read the docs](https://github.com/romkatv/powerlevel10k#manual-font-installation) to learn how to install the needed fonts.
+
+=== "OSX"
+
+    ```bash
+    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}/themes/powerlevel10k
+    sed -i ' ' -e 's/^ZSH_THEME=".*"/ZSH_THEME="powerlevel10k\/powerlevel10k"/g'  ~/.zshrc
+    zsh
+    ```
+
+=== "Linux"
+
+    ```bash
+    git clone --depth=1 https://github.com/romkatv/powerlevel10k.git ${ZSH_CUSTOM:-$HOME/.oh-my-zsh/custom}/themes/powerlevel10k
+    sed 's/^ZSH_THEME=".*"/ZSH_THEME="powerlevel10k\/powerlevel10k"/g' -i ~/.zshrc
+    zsh
+    ```
+
+When opening a ZSH shell for the first time (done by calling `zsh` above), a menu will appear and ask some questions to setup the shell to your convenience.
+
+This setup phase can be restarted anytime with the command `p10k configure`
+
+Here are the answers to get a cool but productive shell:
+
+- `prompt style`: `rainbow` is generally the best look(3)
+- `Character Set`: `Unicode` (1)
+- `Show current time`: use 24h (2)
+- `Prompt Separators`: Angled (1)
+- `Prompt Heads`: Sharp (1)
+- `Prompt Tails`: Flat (1)
+- `Prompt Height`: using only one line is better when copy/pasting (1)
+- `Prompt Connection`: (if `Prompt Height` is set to 2): Disconnected (1)
+- `Prompt Frame`: (if `Prompt Height` is set to 2): No Frame (1)
+- `Prompt Spacing`: Compact (1)
+- `Icons`: Many (2)
+- `Prompt Flow`: Concise (1)
+- `Enable Transient Prompt?`: Yes (y)
+- `Instant Prompt Mode`: speed up slow scripts... set to Verbose as recommended (1)
+- `Apply changes to ~/.zshrc`: Damn YES (y)
+
+#### Plugins
+
+ZSH uses plugins that extend the shortcuts and other behaviours for default apps. Here's the list I usually use:
+
+```yaml
```

Comment:
Why Vault and terraform here? 

Can we add following instead:
`zsh-autosuggestions zsh-syntax-highlighting kube-ps1 kubectl gcloud`

Reply:
kube-ps1 is already taken care by powerLevel10k

looking into the others, but I guess they are too...